### ISSUE TITLE
SetCookieHeaderValue - System.TypeLoadException: Could not load type Microsoft.Extensions.Primitives.InplaceStringBuilder'

### ISSUE THREAD

**Issue by joaocpaiva** [dotnet/runtime#64786]
### Description

This API is raising an exception because it cannot find type Microsoft.Extensions.Primitives.InplaceStringBuilder.

This happens both on .NET Framework 4.7.2 as well as .NET 6.0. I tried reference Microsoft.Net.Http.Headers 2.2.8, but the same exception is raised.

#region Assembly Microsoft.Net.Http.Headers, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60
// D:\Git\AD-AggregatorService-Common\packages\microsoft.net.http.headers\2.2.0\lib\netstandard2.0\Microsoft.Net.Http.Headers.dll
// Decompiled with ICSharpCode.Decompiler 6.1.0.5902
#endregion

Test method Microsoft.Identity.Services.Http.UnitTest.CookieTest.Cookie_HeaderRoundtripping_Test threw exception: 
System.TypeLoadException: Could not load type 'Microsoft.Extensions.Primitives.InplaceStringBuilder' from assembly 'Microsoft.Extensions.Primitives, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'.

Stack Trace: 
    SetCookieHeaderValue.ToString()
    Cookie.ToSetCookieHeader() line 115
    CookieTest.Cookie_HeaderRoundtripping_Test(String originalHeader, String expectedHeader) line 33

### Reproduction Steps
```c#
        internal string ToSetCookieHeader()
        {
            SetCookieHeaderValue header = new SetCookieHeaderValue(this.Name, this.Value);
            header.Domain = this.Domain;
            header.Path = this.Path;
            header.MaxAge = this.MaxAge;
            header.Expires = this.Expires;
            header.HttpOnly = this.HttpOnly;
            header.Secure = this.Secure;
            header.SameSite = ConvertSameSiteModeAppToFramework(this.SameSite);

            return header.ToString();
        }
```
### Expected behavior

SetCookieHeaderValue.ToString() should not throw.

### Actual behavior

SetCookieHeaderValue.ToString() raises TypeLoadException.

### Configuration

.NET Framework 4.7.2 and .NET 6

Windows 11, x64

**Comment by ManickaP** [comment]
What type of project do you have? It works for me with `<Project Sdk="Microsoft.NET.Sdk.Web">`.
Is it possible you have console app and thus all the asp net core libs are not loaded?

**Comment by joaocpaiva** [comment]
This is running as <Project Sdk="Microsoft.NET.Sdk"> with MSTest SDK

    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.0.0" />
    <PackageReference Include="MSTest.TestAdapter" Version="2.2.8" />
    <PackageReference Include="MSTest.TestFramework" Version="2.2.8" />

Why can't the underlying API find InplaceStringBuilder, both on Net472 and Net6.0?

**Comment by davidfowl** [comment]
we delete that API a while ago. That's' why it doesn't work.

**Comment by joaocpaiva** [comment]
@davidfowl I think you mean InplaceStringBuilder?

But if I am using **SetCookieHeaderValue** from **Microsoft.Net.Http.Headers.dll** shouldn't the API work, instead of throwing an exception?

**Comment by madelson** [comment]
This issue is pretty annoying because it means you can run into problems whenever you pair one of the older packages with a newer version of Microsoft.Extensions.Primitives. 

This situation is relatively easy to get yourself into if you have libraries that need to support .NET Framework and .NET Core, since .NET Framework only works with the older ASP.NET Core 2.x packages which rely on `InPlaceStringBuilder`.

If the `InPlaceStringBuilder` API had been retained or could be restored it would have (and still could) save us a lot of pain. Seems like it could just be marked as `Obsolete`.

**Comment by joaocpaiva** [comment]
That's exactly the case.

@davidfowl any chances we could retain **InPlaceStringBuilder**, mark it as Obsolete instead? A quick search online finds us many threads about typeloadexception on **InPlaceStringBuilder** type ... That would solve a lot of problems to the community.

**Comment by karelz** [comment]
Triage: Looks like trouble in using combo of packages where they should not be. Or some ASP.NET packages might need to stay more backward compatible.
Closing here as it is unrelated to Runtime.

**Comment by davidfowl** [comment]
Not quite. If we restore this type it would be a change in Microsoft.Extesnsions.Primitives.

cc @eerhardt

**Comment by eerhardt** [comment]
Is `Microsoft.Net.Http.Headers` even a supported package anymore? I don't see any updates to it in over 2 years.

I'm not inclined to add this API back in order to support an EOL ASP.NET package. If something needs to change here, `Microsoft.Net.Http.Headers` should be updated to the latest `Microsoft.Extensions.Primitives` version. I can transfer this issue to the dotnet/aspnetcore repo.

**Comment by eerhardt** [comment]
See also https://github.com/dotnet/extensions/issues/3800 which is a duplicate and resolved as "Won't fix"

**Comment by joaocpaiva** [comment]
> 

Is there an equivalent class to **SetCookieHeaderValue** with same ToString() logic elsewhere?

**Comment by eerhardt** [comment]
> Is there an equivalent class to SetCookieHeaderValue with same ToString() logic elsewhere?

I believe it still exists, it is located here:

https://github.com/dotnet/aspnetcore/blob/e371b5d1601f0e7c61b4cd94155d5cd2fa3f8d8d/src/Http/Headers/src/SetCookieHeaderValue.cs#L182-L184

If you are a class library and want to use this ASP.NET API, check out https://docs.microsoft.com/en-us/aspnet/core/fundamentals/target-aspnetcore?view=aspnetcore-6.0&tabs=visual-studio. If you are in an asp.net application, you should just update to the latest version (.NET 6) and it will be available.
